aten/src/ATen/native/native_functions.yaml

```diff
@@ -1097,13 +1097,13 @@
   dispatch:
     MkldnnCPU: mkldnn_linear
 
-- func: fbgemm_linear_int8_weight(Tensor input, Tensor weight, Tensor packed, Tensor col_offsets, Scalar weight_scale, Scalar weight_zero_point, Tensor bias) -> Tensor
+- func: fbgemm_linear_int8_weight_fp32_activation(Tensor input, Tensor weight, Tensor packed, Tensor col_offsets, Scalar weight_scale, Scalar weight_zero_point, Tensor bias) -> Tensor
```

Comment:
To preserve backward-compatibility, I suggest the following:

1) Keep `fbgemm_linear_{int8,fp16}_weight` functions.
2) Create separate `fbgemm_linear_{int8,fp16}_weight_fp32_activation` schema and functions.
3) Move implementations from the original functions into the `_fp32_activation` functions.
4) Have the original functions call directly into the `_fp32_activation` functions, and add a TORCH_WARN() call warning that those functions are deprecated

Reply:
Once we can ensure that all internal users have migrated off the old function names, we can then delete the old function schemas and implementations

Reply:
Thanks @jamesr66a ! Updated the code with 1, 2, 3, 4 according to your suggestions.